services/gps2tsdb/gps.py

```diff
@@ -4,18 +4,29 @@
 import dbus
 import postgres
 import os
+import sys
+import logging
 from prometheus_client import start_http_server, Gauge
 
+# Accepts strings and integers
+env_log_level = os.environ.get('LOGLEVEL', 'WARNING').upper()
+log_level = int(env_log_level) if env_log_level.isdigit() else env_log_level
+logging.basicConfig(
+    level=log_level,
+    format='%(asctime)s.%(msecs)03d UTC %(levelname)-8s %(message)s',
+    datefmt='%Y-%m-%d %H:%M:%S')
+
 def fix(*args):
-    #print(args)
-    print("Time: ", args[0], "\tLat: ", args[3], "\tLng: ", args[4])
+    logging.info("Time: ", args[0], "\tLat: ", args[3], "\tLng: ", args[4])
     lat_gauge.set(args[3])
     lng_gauge.set(args[4])
     time_gauge.set(args[0])
 
-    print("Inserting lat and lng for timestanp ", args[0])
+    logging.info("Inserting lat and lng for timestanp ", args[0])
     db.run("INSERT INTO gps (time, lat, lng) VALUES (to_timestamp(%s), %s, %s)", (float(args[0]), float(args[3]), float(args[4]) ) )
-    print("Finsihed inserting lat and lng for timestamp ", args[0])
+    logging.info("Finsihed inserting lat and lng for timestamp ", args[0])
+
+    sys.stdout.flush()
```

Comment:
When running on on of our test machines, I get the following error:
 ```
gps2tsdb_1  | --- Logging error ---
gps2tsdb_1  | Traceback (most recent call last):
gps2tsdb_1  |   File "/usr/local/lib/python3.8/logging/__init__.py", line 1081, in emit
gps2tsdb_1  |     msg = self.format(record)
gps2tsdb_1  |   File "/usr/local/lib/python3.8/logging/__init__.py", line 925, in format
gps2tsdb_1  |     return fmt.format(record)
gps2tsdb_1  |   File "/usr/local/lib/python3.8/logging/__init__.py", line 664, in format
gps2tsdb_1  |     record.message = record.getMessage()
gps2tsdb_1  |   File "/usr/local/lib/python3.8/logging/__init__.py", line 369, in getMessage
gps2tsdb_1  |     msg = msg % self.args
gps2tsdb_1  | TypeError: not all arguments converted during string formatting
gps2tsdb_1  | Call stack:
gps2tsdb_1  |   File "./gps.py", line 64, in <module>
gps2tsdb_1  |     loop.run()
gps2tsdb_1  |   File "/opt/venv/lib/python3.8/site-packages/gi/overrides/GLib.py", line 497, in run
gps2tsdb_1  |     super(MainLoop, self).run()
gps2tsdb_1  |   File "/opt/venv/lib/python3.8/site-packages/dbus/connection.py", line 537, in _signal_func
gps2tsdb_1  |     match.maybe_handle_message(message)
gps2tsdb_1  |   File "/opt/venv/lib/python3.8/site-packages/dbus/connection.py", line 232, in maybe_handle_message
gps2tsdb_1  |     self._handler(*args, **kwargs)
gps2tsdb_1  |   File "./gps.py", line 21, in fix
gps2tsdb_1  |     logging.info("Time: ", str(args[0]), "\tLat: ", str(args[3]), "\tLng: ", str(args[4]))
gps2tsdb_1  | Message: 'Time: '
gps2tsdb_1  | Arguments: ('1601867680.0', '\tLat: ', '40.429996107128', '\tLng: ', '-86.91218787641982')
gps2tsdb_1  | --- Logging error ---
gps2tsdb_1  | Traceback (most recent call last):
gps2tsdb_1  |   File "/usr/local/lib/python3.8/logging/__init__.py", line 1081, in emit
gps2tsdb_1  |     msg = self.format(record)
gps2tsdb_1  |   File "/usr/local/lib/python3.8/logging/__init__.py", line 925, in format
gps2tsdb_1  |     return fmt.format(record)
gps2tsdb_1  |   File "/usr/local/lib/python3.8/logging/__init__.py", line 664, in format
gps2tsdb_1  |     record.message = record.getMessage()
gps2tsdb_1  |   File "/usr/local/lib/python3.8/logging/__init__.py", line 369, in getMessage
gps2tsdb_1  |     msg = msg % self.args
gps2tsdb_1  | TypeError: not all arguments converted during string formatting
gps2tsdb_1  | Call stack:
gps2tsdb_1  |   File "./gps.py", line 64, in <module>
gps2tsdb_1  |     loop.run()
gps2tsdb_1  |   File "/opt/venv/lib/python3.8/site-packages/gi/overrides/GLib.py", line 497, in run
gps2tsdb_1  |     super(MainLoop, self).run()
gps2tsdb_1  |   File "/opt/venv/lib/python3.8/site-packages/dbus/connection.py", line 537, in _signal_func
gps2tsdb_1  |     match.maybe_handle_message(message)
gps2tsdb_1  |   File "/opt/venv/lib/python3.8/site-packages/dbus/connection.py", line 232, in maybe_handle_message
gps2tsdb_1  |     self._handler(*args, **kwargs)
gps2tsdb_1  |   File "./gps.py", line 26, in fix
gps2tsdb_1  |     logging.info("Inserting lat and lng for timestanp ", str(args[0]))
gps2tsdb_1  | Message: 'Inserting lat and lng for timestanp '
gps2tsdb_1  | Arguments: ('1601867680.0',)
```

Unfortunately this appears to mean that `logging.xyz() ` is not a drop in replacement for `print()` We would either have to use [string formatting](https://docs.python.org/3/library/logging.html#logging.Logger.debug) or just concatenate everything (`logging.info("Time: " + str(args[0]) + "\tLat: " + str(args[3]) + "\tLng: " + str(args[4]))`). String concatenation was successfully tested on `dev05`